src/tds/challenge.c

```diff
@@ -174,13 +174,15 @@ make_ntlm_v2_hash(TDSSOCKET * tds, const char *passwd, unsigned char ntlm_v2_has
 	user_name = tds_dstr_cstr(&tds->login->user_name);
 
 	/* parse domain\username */
-	p = strchr(user_name, '\\');
+	p = user_name ? strchr(user_name, '\\') : NULL;
 
 	domain = user_name;
-	domain_len = p - user_name;
+	domain_len = p ? p - user_name : 0;
 
-	user_name = p + 1;
-	user_name_len = strlen(user_name);
+	if (p != NULL) {
+		user_name = p + 1;
+		user_name_len = strlen(user_name);
+	}
```

Comment:
useless too, although in this case compiler can't much help, the username here has to have that kind of format

Reply:
I was concerned that the (indirect) caller might have supplied an unsuitably formatted username, but a closer look reveals that  `tds7_send_auth`  checks for a little-endian UCS-2 backslash before kicking off the call chain that leads to `make_ntlm_v2_hash`.